packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -800,8 +729,6 @@ class HotRunner extends ResidentRunner {
       }
     }
     assert(reassembleViews.isNotEmpty);
-    printTrace('Evicting dirty assets');
-    await _evictDirtyAssets();
```

Comment:
Why is this getting deleted? Does flutter engine no longer need to be told to refresh assets when they are changed?

Reply:
As far as I can tell we only support updating assets on hot restart. These calls are fired on every hot reload regardless of the changed files and seem to serve no purpose

Reply:
I know that you can use hot reload when you (just) update assets. Well, at least it used to work in the past fine.
It's good idea to skip sending this request if no assets were changed though.

Reply:
Its possible we regressed or removed this feature in the past, or I might be missing something. I will do some archaeology tomorrow to confirm

Reply:
I looked into this some more, we support reloading assets but only in the case where the content of the asset itself has changed - we don't support adding/removing/renaming them.

I've added back the evict call, but we can still skip scanning the bundle since additional/removed assets wont show up until a restart anyway

Reply:
Supposedly we need to scan assets to figure out which ones were changed, which ones need to be transferred to the device.